zipline/algorithm.py

```diff
@@ -303,16 +299,14 @@ def __init__(self, *args, **kwargs):
             get_calendar("NYSE")
         )
 
-        # set the capital base
-        self.capital_base = kwargs.pop('capital_base', DEFAULT_CAPITAL_BASE)
         self.sim_params = kwargs.pop('sim_params', None)
         if self.sim_params is None:
             self.sim_params = create_simulation_parameters(
-                capital_base=self.capital_base,
                 start=kwargs.pop('start', None),
                 end=kwargs.pop('end', None),
                 trading_calendar=self.trading_calendar,
             )
+        self.capital_base = self.sim_params.capital_base
```

Comment:
Looks like this is only used for the `__repr__` now. Could we kill it and have that method reference `self.sim_params.capital_base` instead, so we have one less piece of state to keep track of?